File: src/main/java/fi/aalto/cs/apluscourses/intellij/notifications/MissingModuleNotification.java
package fi.aalto.cs.apluscourses.intellij.notifications;

import com.intellij.notification.Notification;
import com.intellij.notification.NotificationType;
import fi.aalto.cs.apluscourses.intellij.services.PluginSettings;
import org.jetbrains.annotations.NotNull;

import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getAndReplaceText;
import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getText;

public class MissingModuleNotification extends Notification {

  @NotNull
  private final String moduleName;

  /**
   * Construct a missing module notification that explains that a module with the given name
   * couldn't be found.
   */
  public MissingModuleNotification(@NotNull String moduleName) {
    super(
        PluginSettings.A_PLUS,
        getText("notification.MissingModuleNotification.title"),
        getAndReplaceText("notification.MissingModuleNotification.content", moduleName),
        NotificationType.ERROR);
    this.moduleName = moduleName;
  }

  @NotNull
  public String getModuleName() {
    return moduleName;
  }

}

File: src/main/kotlin/fi/aalto/cs/apluscourses/notifications/MissingModuleNotification.kt
package fi.aalto.cs.apluscourses.notifications

import com.intellij.notification.Notification
import com.intellij.notification.NotificationType
import fi.aalto.cs.apluscourses.MyBundle
import fi.aalto.cs.apluscourses.services.PluginSettings

/**
 * Construct a missing module notification that explains that a module couldn't be found.
 */
class MissingModuleNotification : Notification(
    PluginSettings.A_PLUS,
    MyBundle.message("notification.MissingModuleNotification.title"),
    MyBundle.message("notification.MissingModuleNotification.content"),
    NotificationType.ERROR
)
